me.id.webverify/webverifylib/src/main/java/me/id/webverifylib/IDmeWebVerify.java

```diff
@@ -98,24 +112,21 @@ public void getAccessToken(IDmeScope scope, boolean forceReload, IDmeGetAccessTo
     throw new UnsupportedOperationException();
   }
 
-  public void getUserProfile(IDmeGetProfileListener listener) {
-    checkInitialization();
-    // TODO mirland 13/12/16:
+  public void getUserProfile(IDmeScope scope, IDmeGetProfileListener listener) {
+    AuthToken token = accessTokenManager.getToken(scope);
+    if (token == null) {
+      String message = String.format(Locale.US, "There is not an access token related to the %s scope", scope);
+      listener.onError(new IllegalStateException(message));
+    } else if (token.isValidToken()) {
+      String requestUrl = createRequestUrl(token.getAccessToken());
+      new GetProfileConnectionTask(listener).execute(requestUrl);
+    } else {
+      listener.onError(new IllegalStateException("The access token is expired"));
+    }
   }
 
   public void logOut() {
-    checkInitialization();
-    // TODO mirland 13/12/16:
-  }
-
-  /**
-   * Checks if the application is already initialized
-   * @throws IllegalStateException
-   */
-  private void checkInitialization() {
-    if (!initialized) {
-      throw new IllegalStateException("IDmeWebVerify has to be initialized before use any operation");
-    }
+    accessTokenManager.deleteSession();
```

Comment:
ok, we can do that, but in my the logout function should exist. What do you think @mats-claassen ?